expression/builtin_time.go

```diff
@@ -7111,5 +7111,5 @@ func handleInvalidZeroTime(ctx sessionctx.Context, t types.Time) (bool, error) {
 	if isOriginalIntOrDecimalZero && !isOriginalStringZero {
 		return false, nil
 	}
-	return true, handleInvalidTimeError(ctx, types.ErrWrongValue.GenWithStackByArgs(types.DateTimeStr, t.String()))
+	return false, handleInvalidTimeError(ctx, types.ErrWrongValue.GenWithStackByArgs(types.DateTimeStr, t.String()))
```

Comment:
I saw some built-in functions (such as DayOfYear) also called `handleInvalidZeroTime` without checking `NO_ZERO_DATE`. Is the result still expected in this case?

Reply:
I know. And actually I am confused about here. I don't know why we need to return NULL when year(0.9) but return 0 when year(0)